lib/alchemy/config_loader.rb

```diff
@@ -0,0 +1,85 @@
+module Alchemy
+  class ConfigLoader
+    attr_reader :which
+
+    def initialize(which)
+      @which = which.sub(/\.yml$/, '')
+    end
+
+    # Reads the config file named +<which>.yml+ from +config/alchemy/+ folder.
+    #
+    def load_all
+      if paths.empty?
+        raise LoadError, "Could not find #{which}.yml file! Please run `rails generate alchemy:scaffold`"
+      else
+        # OPTIMIZE: remove duplicates?
+        paths.map(&method(:load_file)).inject(&method(:merge))
+      end
+    end
+
+    # Returns the path to +<which>.yml+ file
+    #
+    def file_path
+      Rails.root.join relative_path
+    end
+
+    def file_name
+      "#{which}.yml"
+    end
+
+    def relative_path
+      Pathname.new('config').join('alchemy', file_name)
+    end
+
+    def paths
+      @paths ||= candidates.select(&:exist?)
+    end
+
+    private
+
+    def load_file(file_path)
+      ::YAML.load(ERB.new(File.read(file_path)).result) || default
+    end
+
+    def merge(a, b)
+      if which.ends_with?('s')
```

Comment:
`elements.yml` and `page_layouts.yml` represent an Array, `config.yml` a hash. I thought about giving the merger on construction, but this looked more complicated.